### ISSUE TITLE
Automatic-Mask-Generation use finetuning model on SAM.

### ISSUE THREAD

**Issue by unmo** [NielsRogge/Transformers-Tutorials#443]
I followed [this notebook](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/SAM/Fine_tune_SAM_(segment_anything)_on_a_custom_dataset.ipynb) for finetuning the SAM on custom data, and save finetuning model to "./model_path".

The following code was executed using the resulting model.

```
from transformers import pipeline
generator = pipeline("mask-generation", model="./model_path", device=0, output_bboxes_mask=True)
outputs = generator(img, points_per_batch=1)
```

However, I got the same result as the following code.
```
from transformers import pipeline
generator = pipeline("mask-generation", model="facebook/sam-vit-large", device=0, output_bboxes_mask=True)
outputs = generator(img, points_per_batch=1)
```

Can you tell me why this happens?

**Comment by NielsRogge** [comment]
Hi,

Perhaps you can check some weights using `generator.model` and see whether you're actually using your custom model.

**Comment by pzhren** [comment]
hi, I would like to know if sam in this transformer library supports Automatic-Mask-Generation (everything segment)? How to achieve it?

**Comment by pzhren** [comment]
hi, 
how can the fine-tuned saved model be reloaded in the pipeline method? Thank you very much
![image](https://github.com/user-attachments/assets/46a45211-aff1-4077-928f-eee03e5bece8)
